docs/design/datacontracts/Thread.md

```diff
@@ -0,0 +1,174 @@
+# Contract Thread
+
+This contract is for reading and iterating the threads of the process.
+
+## Data structures defined by contract
+``` csharp
+record struct DacThreadStoreData (
+    int ThreadCount,
+    int UnstartedThreadCount,
+    int BackgroundThreadCount,
+    int PendingThreadCount,
+    int DeadThreadCount,
+    TargetPointer FirstThread,
+    TargetPointer FinalizerThread,
+    TargetPointer GcThread);
+
+enum ThreadState
+{
+    TS_Unknown                = 0x00000000,    // threads are initialized this way
```

Comment:
We should review these flags. Number of them do not make sense to include as a contract.

Reply:
Good idea.